app/jobs/build_job.rb

```diff
@@ -25,6 +26,7 @@ def perform
       build_config(dir).write("#{dir}/bt.sh")
       return short_circuit! if short_circuit?(run_docker_container(dir))
       set_status(:success)
+      set_completed_at
```

Comment:
I think we should do one of four things here.
1. Copy this line to just after `set_status(:error)` and to the `short_circuit!` method below such that duration gets set even if thre's an error or a failure.
2. Integrate this into `set_status` such that on changes to non `:pending` statuses we set completed_at
3. Drop an afterhook into the build model to set completed_at.
4. Drop this into an ensure block in `perform` so it happens at the end regardless.